### PR TITLE
use rhel-9-golang-1.20 builder image

### PR DESCRIPTION
**Pull request by andfasano**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [X] CI/CD 

## What environments does this code impact?

- [X] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [X] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by andfasano** [comment]
/hold

Requires https://github.com/openshift/release/pull/41331

File: Dockerfile.assisted-service.ocp [hunk 1]
```diff
@@ -1,8 +1,4 @@
-# URGENT! ART metadata configuration has a different number of FROMs
-# than this Dockerfile. ART will be unable to build your component or
-# reconcile this Dockerfile until that disparity is addressed.
-# Build binaries
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.14 AS builder
```
Comment (osherdp) [inline Dockerfile.assisted-service.ocp:1]: is this PR blocked by https://github.com/openshift/release/pull/41331 or we just don't want prow to garbage-collect the ``ocp_builder_rhel-9-golang-1.20-openshift-4.14`` images?

Comment (andfasano) [reply Dockerfile.assisted-service.ocp:1]: Yes I've added a comment for that https://github.com/openshift/assisted-service/pull/5361#issuecomment-1637847768

**Comment by andfasano** [comment]
Requires also https://issues.redhat.com/browse/ART-7241 for the nmstate dependencies

File: Dockerfile.assisted-service.ocp (hunk 1 above)
[the remark this answers is not in the document]
Comment (andfasano) [reply Dockerfile.assisted-service.ocp:1]: Right, for that https://issues.redhat.com/browse/ART-7241 is required

Comment (osherdp) [reply Dockerfile.assisted-service.ocp:1]: Yeah, saw that comment. Just wasn't sure for the exact reason (which of the two reasons mentioned above applies here?)

Comment (andfasano) [reply Dockerfile.assisted-service.ocp:1]: The first one, waiting to have the new image available in the CI config so that we could test the switch here

Comment (andfasano) [reply Dockerfile.assisted-service.ocp:1]: (and btw the bot didn't update as expected the Dockerfile)

Comment (andfasano) [reply Dockerfile.assisted-service.ocp:1]: @osherdp I've double checked and given the current configuration should be ok to land this PR first, in any case it's blocked by  https://issues.redhat.com/browse/ART-7241

Comment (osherdp) [reply Dockerfile.assisted-service.ocp:1]: yes that makes sense as the changes in release repo are mainly image substitutions for performance reasons iiuc
either way I'll start following that ticket to be notified for the follow-ups

**Comment by andfasano** [comment]
Will be addressed in https://issues.redhat.com/browse/CORS-2750

**Pull request closed without merging**